### PR TITLE
feat: account list

### PR DESCRIPTION
**Pull request by davidyuk**
Screenshot 2021-03-16 at 09 19 51

This should be enough to implement corresponding designs.

### CODE REVIEW

**Review by mradkov — COMMENTED** [review]
Code looks good. Do we keep track of the selected account across other components e.g. payments/tips?

**Comment by davidyuk** [comment]
@mradkov The idea is to have a single source of account data, and the account switcher updates it, so all components should display actual data. Currently, in some cases like the aepp-wallet connection, it may not work properly, I'm planning to check and fix it later.

File: src/popup/router/components/AccountCard.vue [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  border-radius: 6px;
+  // stylelint-disable-next-line declaration-colon-newline-after
+  background: url('../../../icons/account-card/account-bg-pattern.svg'),
+    linear-gradient(rgb(10, 14, 22), rgb(10, 14, 22));
+  background-position: calc(var(--shift) * 15px) calc(var(--shift) * 20px), center, center, center;
+
+  &.subaccount {
+    // stylelint-disable-next-line declaration-colon-newline-after
+    background: url('../../../icons/account-card/subaccount-bg-pattern.svg'),
+      linear-gradient(#131313, #131313);
```
Comment (davidyuk) [inline src/popup/router/components/AccountCard.vue:64]:
```suggestion
      $color-bg-1;
```

File: src/popup/router/components/AccountCard.vue [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    background-position: calc(var(--shift) * 15px) calc(var(--shift) * 20px), center, center, center;
+  }
+
+  &.first-subaccount.subaccount {
+    // stylelint-disable-next-line declaration-colon-newline-after
+    background: url('../../../icons/account-card/subaccount-bg-pattern.svg'),
+      linear-gradient(rgb(19, 27, 42), rgb(19, 27, 42));
```
Comment (davidyuk) [inline src/popup/router/components/AccountCard.vue:71]:
```suggestion
    background:
      url('../../../icons/account-card/subaccount-bg-pattern.svg'),
      linear-gradient(rgb(19, 27, 42), rgb(19, 27, 42));
```
https://github.com/stylelint/stylelint-config-standard/blob/bc51daa35a19e5cdb32188a22a07e94bb02be496/index.js#L47
https://stylelint.io/user-guide/rules/declaration-colon-newline-after#always-multi-line

File: src/popup/router/components/AccountInfo.vue [hunk 3]
```diff
@@ -1,115 +1,307 @@
...
+      await this.$store.dispatch('modals/open', {
+        name: 'confirm',
+        title: this.$t('modals.removeSubaccount.title'),
+        msg: this.$t('modals.removeSubaccount.msg'),
+      });
+      this.deleteAccount(this.idx);
+    },
+    openAccountInExplorer() {
+      openUrl(
+        `https://explorer.aeternity.io/account/transactions/${this.accounts[this.idx].address}`,
```
Comment (davidyuk) [inline src/popup/router/components/AccountInfo.vue:109]: use `getExplorerPath` getter

File: src/popup/router/components/SidebarMenu.vue [hunk 4]
```diff
@@ -10,7 +10,7 @@
         <div class="f-12" v-if="account.name" data-cy="chain-name">
           <TruncateMid :str="account.name" />
         </div>
-        <div class="account-type">{{ $t('mainAccount') }}</div>
+        <div class="account-type">{{ account.userName }}</div>
```
Comment (davidyuk) [inline src/popup/router/components/SidebarMenu.vue:13]: `userName` is confusing, it sounds like the account is referencing some user by name. Also, aens names are also defined by the user. Proposing to use `localName` instead.

File: src/store/mutations.js [hunk 5]
```diff
@@ -96,4 +97,30 @@ export default {
...
     state.loginTargetLocation = location;
   },
+  createAccount(state) {
+    state.accs.push({
+      idx: state.accountCount,
+      userName: i18n.t('subaccountName', { number: state.accountCount }),
+      color:
+        // eslint-disable-next-line no-bitwise
+        state.accountCount === 1 ? '#00FF9D' : `#${((Math.random() * 0xffffff) << 0).toString(16)}`,
+      shift: Math.floor(Math.random() * 100),
```
Comment (davidyuk) [inline src/store/mutations.js:107]: let's generate them based on account address as I proposed [here](https://github.com/aeternity/aepp-sdk-js/pull/65/files#diff-b088c63374aa5e83e46a7d26a08f04b14d1918a1fdb885f003b9d98e64300816R23-R24), to have them recoverable by mnemonic phrase

File: src/store/mutations.js [hunk 6]
```diff
@@ -96,4 +97,30 @@ export default {
   setLoginTargetLocation(state, location) {
     state.loginTargetLocation = location;
   },
+  createAccount(state) {
+    state.accs.push({
+      idx: state.accountCount,
+      userName: i18n.t('subaccountName', { number: state.accountCount }),
```
Comment (davidyuk) [inline src/store/mutations.js]: More accurate would be to add this in `accounts` getter (`userName || i18n.t(...)`) to keep it translatable and be able to change the wording in the future.

File: src/popup/router/components/AccountSwitcher.vue [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+import { mapState, mapGetters } from 'vuex';
+import AccountCard from './AccountCard';
+
+export default {
+  components: { AccountCard },
+  props: { notification: Boolean },
+  data: () => ({
+    selectedCardNumber: 0,
+  }),
+  computed: {
```
Comment (davidyuk) [inline src/popup/router/components/AccountSwitcher.vue:36]:
```suggestion
  computed: {
    selectedCardNumber() {
      return this.filteredAccounts.findIndex(
       (a) => a.i === this.accountSelectedIdx,
      );
    },
```

File: src/popup/router/components/AccountCard.vue (hunk 2 above)
Comment (CedrikNikita) [reply src/popup/router/components/AccountCard.vue:69]: prettier and stylelint are conflicting in this place, that is the reason i used this disabling comment

File: src/popup/router/components/AccountSwitcher.vue (hunk 7 above)
Comment (CedrikNikita) [reply src/popup/router/components/AccountSwitcher.vue]: Yes, missed that somehow

File: src/popup/router/components/AccountInfo.vue (hunk 3 above)
Comment (CedrikNikita) [reply src/popup/router/components/AccountInfo.vue:109]: The links are not the same

**Comment by mradkov** [comment]
Screenshot 2021-03-24 at 11 22 16

Cannot generate account.

**Comment by Liubov-crypto** [comment]
Web wallet: 
Add new sub account - pass
Balances - pass 
Search tokens - pass 
Activity History - pass 
Sort by Date - pass 
Info about transaction - pass
Copy wallet address - pass 
Change currency - pass 
Change tokens - pass 
Send Tokens - pass, using QR - pass 
Send Tip - pass 
Retip - pass 
Register a name - pass 
Extend - pass 
Set name pointer - pass 
Set as default name - pass 
Generate invite link - pass 
Claim back invite link - failed  
Top Up invite link - pass 
Settings: 
Change language - pass 
Security&Privacy - pass 
Change Network - pass 
Permission Settings - pass 
Remove account - pass 

FF/Chrome extension:
Add new sub-account - pass
Balances - pass 
Search tokens - pass 
Activity History - pass 
Sort by Date - pass 
Info about transaction - pass
Copy wallet address - pass
Change currency - pass 
Change tokens - pass 
Send Tokens - pass, using QR - pass 
Send Tip - pass 
Retip - pass 
Register a name - pass 
Extend - pass 
Set name pointer - pass 
Set as default name - pass 
Generate invite link - pass 
Claim back invite link - failed  
Top Up invite link - pass 
Settings: 
Change language - pass 
Security&Privacy - pass 
Change Network - pass 
Permission Settings - pass 
Remove account - pass 

Dropdown menu bug: 
drop-down bug
drop-down menu

Not all Fungible tokens are always available:
on 892

in the development branch, I see them all:
АЕ щт вумудщз

The inscription is shifted in the transaction history:
надпись смещена

The extension does not connect with Jelly Swap and Superhero UI.

Got this error when claiming tokens from an invite link:
error when claiming tips  Json parse
json parse

When switching to the testnet: 
errr on the testnet
eeerrr
Could this be related to #518 ? 

When clicking on names in the testnet: 
err with names in the testnet

and trying to extend:
errr with names when extended??

t tx is undefined

**Comment by mradkov** [comment]
@CedrikNikita I'd suggest to merge this in develop and address all of the additional features/functions in separate prs.

**Comment by Liubov-crypto** [comment]
.chain names in the sub-accounts are not automatically displayed, I need to go to the names tab to make it visible in the sub-account. Although .chain name is set by default:

https://user-images.githubusercontent.com/69896204/112527175-6fd3af00-8dab-11eb-8ee6-18f136dc29c7.mov

**Review by davidyuk — COMMENTED** [review]
Code changes look ok

**Review by etharner — APPROVED** [review]

**Pull request merged**